integrations/huggingface_hub/scripts/push_to_hub.py

```diff
@@ -0,0 +1,179 @@
+import os
+import typer
+import tempfile
+import tarfile
+import shutil
+import json
+import yaml
+from huggingface_hub import Repository, HfApi, HfFolder
+
+token_classification_components = ["ner", "tagger", "morphologizer"]
+text_classification_components = ["textcat", "textcat_multilabel"]
+
+def _insert_value(metadata, name, value):
+    if value is None or value == "":
+        return metadata
+    metadata[name] = value
+    return metadata
+
+def _insert_values_as_list(metadata, name, values):
+    if values is None:
+        return metadata
+    if isinstance(values, str):
+        values = [values]
+    if len(values) == 0:
+        return metadata
+    metadata[name] = values
+    return metadata
+
+def create_metric(name, t, value): 
+    return {
+            "name": name,
+            "type": t,
+            "value": value
+        }
+
+def create_p_r_f_list(precision, recall, f_score):
+    precision = create_metric("Precision", "precision", precision)
+    recall = create_metric("Recall", "recall", recall)
+    f_score = create_metric("F Score", "f_score", f_score)
+
+    return [precision, recall, f_score]
+
+def create_model_index(repo_name, data):
+    model_index = {"name": repo_name}
+
+    results = []
+    if "ents_p" in data:
+        results.append({
+            "tasks": {
+                "name": "NER",
+                "type": "token-classification",
+                "metrics": create_p_r_f_list(data["ents_p"], data["ents_r"], data["ents_f"])
+            }
+        })
+    if "tag_acc" in data:
+        results.append({
+            "tasks": {
+                "name": "POS",
+                "type": "token-classification",
+                "metrics": [create_metric("Accuracy", "accuracy", data["tag_acc"])]
+            }
+        })
+    if "sents_p" in data:
+        results.append({
+            "tasks": {
+                "name": "SENTER",
+                "type": "token-classification",
+                "metrics": create_p_r_f_list(data["sents_p"], data["sents_r"], data["sents_f"])
+            }
+        })
+    if "dep_uas" in data:
+        results.append({
+            "tasks": {
+                "name": "UNLABELED_DEPENDENCIES",
+                "type": "token-classification",
+                "metrics": [create_metric("Accuracy", "accuracy", data["dep_uas"])]
+            }
+        })
+    if "dep_las" in data:
+        results.append({
+            "tasks": {
+                "name": "LABELED_DEPENDENCIES",
+                "type": "token-classification",
+                "metrics": [create_metric("Accuracy", "accuracy", data["dep_uas"])]
+            }
+        })
+
+    model_index["results"] = results
+    return model_index
+
+
+def create_model_card(repo_name, repo_dir):
+    with open(os.path.join(repo_dir, "meta.json")) as f:
+        data = json.load(f)
+        lang = data["lang"] if data["lang"] != "xx" else "multilingual"
+        lic = data["license"].replace(" ", "-")
```

Comment:
`license` isn't technically a required field, so this is safer:

```suggestion
        lic = data.get("license", "").replace(" ", "-")
```

Reply:
What I saw is that when it's not there we get `"license":""`, so I think this should be fine. Committing for extra safety :)